src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/PossibleDeNovo.java

```diff
@@ -38,48 +40,80 @@
  *
  */
 @DocumentedFeature(groupName=HelpConstants.DOC_CAT_ANNOTATORS, groupSummary=HelpConstants.DOC_CAT_ANNOTATORS_SUMMARY, summary="Existence of a de novo mutation in at least one of the given families (hiConfDeNovo, loConfDeNovo)")
-public final class PossibleDeNovo extends InfoFieldAnnotation {
+public final class PossibleDeNovo extends PedigreeAnnotation {
+    protected final OneShotLogger warning = new OneShotLogger(this.getClass());
+    private final MendelianViolation mendelianViolation;
+    private Set<Trio> trios;
 
+    @VisibleForTesting
     public PossibleDeNovo(final Set<Trio> trios, final double minGenotypeQualityP) {
+        super((Set<String>) null);
         this.trios = Collections.unmodifiableSet(new LinkedHashSet<>(trios));
         mendelianViolation = new MendelianViolation(minGenotypeQualityP);
     }
 
-    /**
-     * This is dummy constructor that will do nothing until https://github.com/broadinstitute/gatk/issues/1880 is addressed
-     */
-    public PossibleDeNovo(){
-        this(Collections.emptySet(), 0);
+    public PossibleDeNovo(final Set<String> founderIds){
+        super(founderIds);
+        mendelianViolation = new MendelianViolation(DEFAULT_MIN_GENOTYPE_QUALITY_P);
     }
 
-    protected final OneShotLogger warning = new OneShotLogger(this.getClass());
+    public PossibleDeNovo(final File pedigreeFile){
+        super(pedigreeFile);
+        mendelianViolation = new MendelianViolation(DEFAULT_MIN_GENOTYPE_QUALITY_P);
+    }
 
+    public PossibleDeNovo(){
+        this((Set<String>) null);
+    }
 
+    @Override
+    void validateArguments(Collection<String> founderIds, File pedigreeFile) {
+        if (pedigreeFile == null) {
+            if ((founderIds != null && !founderIds.isEmpty())) {
+                warning.warn("PossibleDenovo annotation will not be calculated, must provide a valid PED file (-ped). Founder-id arguments cannot be used for this annotation");
+            } else {
+                warning.warn("PossibleDenovo Annotation will not be calculated, must provide a valid PED file (-ped) from the command line.");
+            }
```

Comment:
Is there a reason this shouldn't throw an exception ?

Reply:
See my above comment

Reply:
See my other comment about having two types of annotations and arguments specified

Reply:
Still don't buy it but we'll leave this as is and continue the discussion in the refactoring PR.